opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -310,6 +218,94 @@ private static ExprValue createOpenSearchDateType(Content value, ExprType type)
     return new ExprTimestampValue((Instant) value.objectValue());
   }
 
+  /**
+   * Extend existing mapping by new data without overwrite. Called from aggregation only {@see
+   * AggregationQueryBuilder#buildTypeMapping}.
+   *
+   * @param typeMapping A data type mapping produced by aggregation.
+   */
+  public void extendTypeMapping(Map<String, OpenSearchDataType> typeMapping) {
+    for (var field : typeMapping.keySet()) {
+      // Prevent overwriting, because aggregation engine may be not aware
+      // of all niceties of all types.
+      if (!this.typeMapping.containsKey(field)) {
```

Comment:
Nit: use putIfAbsent() instead. It might be optimized depending on the map implementation.